Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp

```diff
@@ -3184,6 +3220,27 @@ namespace AssetProcessor
                 // Do not process this file yet.  When the job is done it will retrigger processing for this file.
                 return;
             }
+
+            AZ::Outcome<QString, void> topmostSourceAsset = GetFirstTopmostSourceAssetForProductPath(productName);
+
+            if (topmostSourceAsset.IsSuccess())
+            {
+                if (!QFile::exists(topmostSourceAsset.GetValue()))
```

Comment:
You may want to make use of the file state cache to do this check, hitting the disk tends to be more expensive.
```
auto* diskCache = AZ::Interface<IFileStateRequests>()::Get()
if(!diskCache->Exists(topmostSourceAsset.GetValue()))
```

Reply:
Done. Something I found that was slightly concerning. I used three different exist functions in building this: QFile, AZ::IO::FileIOBase::GetInstance()->Exists(fileName), and the IFileStateRequests.

QFile and IFileStateRequests handled it correctly with the path I gave it. AZ::IO::FileIOBase::GetInstance()->Exists(fileName) did not work, it was reporting the file as not existing when calling QFile and IFileStateRequests both said the file did exist. I don't have enough info here to open a separate ticket, but figured I would point this out for now in case anyone else has encountered something similar.